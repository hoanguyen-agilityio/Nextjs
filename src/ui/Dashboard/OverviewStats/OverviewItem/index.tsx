// Third party
import { memo } from 'react';

// Components
import { ChartArrows } from '@/icons';
import clsx from 'clsx';

interface OverviewItemProps {
  value: string | number;
  label: string;
  growth: string | number;
  isLast: boolean;
  isFirst: boolean;
}

const OverviewItem = memo(
  ({ value, label, growth, isLast, isFirst }: OverviewItemProps) => (
    <div
      className={clsx(
        'flex flex-col items-center',
        { 'border-r border-gray-700 pr-6xl': !isLast },
        { 'pl-7': !isFirst },
      )}
    >
      <span className="text-4xl font-semibold mb-2sm">{value}</span>
      <span className="text-base font-semibold text-ratio-200 dark:text-grayBlue-400 mb-2sm">
        {label}
      </span>
      <div className="flex items-center rounded-lg bg-gray-50 py-sm px-m dark:bg-gray-700">
        <ChartArrows width="12px" height="7px" className="mr-1" />
        <span className="text-sm font-bold">{growth}</span>
      </div>
    </div>
  ),
);

OverviewItem.displayName = 'OverviewItem';

export default OverviewItem;
