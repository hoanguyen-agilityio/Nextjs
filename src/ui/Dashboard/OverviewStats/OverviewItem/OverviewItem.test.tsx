import { render } from '@testing-library/react';
import OverviewItem from '.';

describe('OverviewItem Component', () => {
  test('render overview item', () => {
    const { container } = render(
      <OverviewItem
        value="1000"
        label="Total Sales"
        growth="15%"
        isLast={false}
        isFirst={false}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
