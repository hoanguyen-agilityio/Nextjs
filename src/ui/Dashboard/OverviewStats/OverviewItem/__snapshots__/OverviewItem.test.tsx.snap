// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewItem Component render overview item 1`] = `
<div>
  <div
    class="flex flex-col items-center border-r border-gray-700 pr-6xl pl-7"
  >
    <span
      class="text-4xl mb-2sm"
    >
      1000
    </span>
    <span
      class="text-base text-ratio-200 dark:text-grayBlue-400 mb-2sm"
    >
      Total Sales
    </span>
    <div
      class="flex items-center rounded-lg bg-gray-50 py-sm px-m dark:bg-gray-700"
    >
      <svg
        class="mr-1"
        fill="none"
        height="7px"
        viewBox="0 0 13 8"
        width="12px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="fill-black-900 dark:fill-white"
          clip-rule="evenodd"
          d="M0.25 7.22977C0.0880127 7.07217 -0.00341797 6.85577 -0.00341797 6.62976C-0.00341797 6.40375 0.0880127 6.18735 0.25 6.02975L4 2.38776C4.18665 2.20544 4.44373 2.11426 4.70349 2.13834C4.96313 2.16238 5.19922 2.29929 5.34912 2.51276L6.62805 4.37576L9.29199 1.79675H7.90698C7.58948 1.81851 7.28626 1.66144 7.12085 1.38947C6.95544 1.11749 6.95544 0.776001 7.12085 0.504028C7.28626 0.232056 7.58948 0.0749817 7.90698 0.0967407H11.426C11.9023 0.0927429 12.2939 0.471466 12.3059 0.947754L12.306 4.35376C12.306 4.66815 12.1383 4.95868 11.866 5.11587C11.5938 5.27307 11.2583 5.27307 10.986 5.11587C10.7137 4.95868 10.546 4.66815 10.546 4.35376V2.99176L7.10803 6.31976C6.92151 6.50208 6.66443 6.59326 6.40466 6.56918C6.14502 6.54514 5.90894 6.40823 5.75903 6.19476L4.48108 4.33276L1.48914 7.23276C1.14136 7.56485 0.593994 7.56485 0.246094 7.23276L0.25 7.22977Z"
          fill-rule="evenodd"
        />
      </svg>
      <span
        class="text-sm font-bold"
      >
        15%
      </span>
    </div>
  </div>
</div>
`;
